File: code4.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        self.x=nums1 if len(nums1)<=len(nums2) else nums2
        self.y=nums2 if len(nums1)<=len(nums2) else nums1
        self.lenX=len(self.x)
        self.lenY=len(self.y)
        self.leftX=0
        self.rightX=self.lenX
        self.calculateParam()
        flag=self.check()
        while flag!=0:
            if flag==-1:
                self.rightX=self.midX-1
            else :
                self.leftX=self.midX+1
            self.calculateParam()
            flag=self.check()
        return self.findMedian()
    def calculateParam(self):
        self.midX=int((self.leftX+self.rightX)/2)
        self.midY=int((self.lenX+self.lenY+1)/2)-self.midX
        self.valueLeftX= self.x[self.midX-1] if self.midX>0 else -10000000
        self.valueRightX= self.x[self.midX] if self.midX<self.lenX else 10000000
        self.valueLeftY= self.y[self.midY-1] if self.midY>0 else -10000000
        self.valueRightY= self.y[self.midY] if self.midY<self.lenY else 10000000
    def check(self):
        if self.valueLeftX<=self.valueRightY and self.valueLeftY<=self.valueRightX:
            return 0
        if self.valueLeftX>self.valueRightY:
            return -1
        if self.valueRightX<self.valueLeftY:
            return 1
    def findMedian(self):
        if(self.lenX+self.lenY)%2==0:
            return float(max(self.valueLeftX,self.valueLeftY)+min(self.valueRightX,self.valueRightY))/2
        else:
            return max(self.valueLeftX,self.valueLeftY)

s=Solution()
nums1 = [1,2,5]
nums2 = [3,4]
print(s.findMedianSortedArrays(nums1,nums2))


        

File: code4.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> x,y;
        if(nums1.size()<=nums2.size()){
            x=nums1;
            y=nums2;
        }
        else{
            x=nums2;
            y=nums1;
        }
     
        int lenX=x.size();
        int lenY=y.size();
        int leftX,rightX,midX,midY,xl,xr,yl,yr,flag=-2;
        leftX=0;
        rightX=lenX;

        do{
            if(flag==1){
                leftX=midX+1;
            }else if(flag==-1){
                rightX=midX-1;
            }
            midX=(leftX+rightX)/2;
            midY=(lenX+lenY+1)/2-midX;
            xl=midX>0 ? nums1[midX] : -100000;
            xr=midX<lenX ? nums1[midX+1] : 100000;
            yl=midY>0 ? nums2[midY] : -100000;
            yr=midY<lenY ? nums2[midY+1] : 100000;

            cout<<xl<<" "<<xr<<endl;
            cout<<yl<<" "<<yr<<endl;

            if(xl<=yr && xr>=yl) flag=0;
            else if(xl>yr) flag=-1;
            else if(xr<yl) flag=1;
        }while(flag!=0);

        if((lenX+lenY)%2!=0){
            return max(xl,yl);
        }
        else return (double)(max(xl,yl)+min(xr,yr))/2.0;



    }


};

int main(){
  
    Solution s;
    vector<int> num1;
    vector<int> num2;
    num1.push_back(1);
    num1.push_back(3);
    num2.push_back(4);
    cout<<s.findMedianSortedArrays(num1,num2)<<endl;

}
